### ISSUE TITLE
`new MyClass` must use parenthesis

### ISSUE THREAD

**Issue by SpyroTEQ** [kalessil/phpinspectionsea#1456]
Description (screenshot):
IntelliJ checks the number of arguments when doing a method/constructor call, so this:
```php
class X {
public function __construct($x) {
echo $x;
}
}
$x = new X();
```

Will show a "Wrong number of arguments when calling new X()". So you know you need to add a parameter:

```php
$x = new X(1); // OK for intellij and PHP
```

But, PHP allows to call a constructor without the parenthesis. So this is valid in PHP:
```php
$x = new X; // Valid PHP5/7 syntax
```

But since you miss a parameter, you'll have a runtime error (ArgumentCountException). And intellij doesn't detect it (because there is no parenthesis I suppose?!)

Hence it would be neat if `new X` without parenthesis yields a PHP EA error like `new X must use parenthesis` and suggests to replace with `new X()`: then intellij would trigger its normal arguments-count inspection and show you're missing a parameter here.

--

Example
```php
class X {
public function __construct($x) {
echo $x;
}
}
$x = new X; // Constructor call must use parenthesis => suggest new X() and let intellij see (afterward) that the parameter count differ
```

--
I've integrated this into my own plugin, but it might be nice to add it to PHP EA instead:

```java
package com.reinom.intellij.plugin.generic;

import com.intellij.codeInspection.ProblemsHolder;
import com.intellij.psi.PsiElement;
import com.jetbrains.php.lang.psi.elements.NewExpression;
import com.jetbrains.php.lang.psi.visitors.PhpElementVisitor;
import com.reinom.intellij.plugin.inspection.APhpReinomInspection;
import com.reinom.intellij.plugin.inspection.ReinomInspection;
import com.reinom.intellij.plugin.inspection.ReinomLocalQuickFix;
import org.jetbrains.annotations.NotNull;

public class PhpConstructorCallMustUseParenthesisInspection extends APhpReinomInspection {
	protected PhpConstructorCallMustUseParenthesisInspection() {
		super(
				REINOM_PRAWD,
				"Constructor called without parenthesis",
				"Calling 'new MyClass' without parenthesis is valid in PHP5/7" +
						" but it forbids intellij from checking the constructor's call parameters" +
						" hence a 'class MyClass { public function __construct($x) { ... } }'" +
						" called with '$x = new MyClass' would not trigger any intellij inspection," +
						" while PHP screams a legit 'ArgumentCountException' at Runtim." +
						"\nHence forcing the use of parenthesis (intellij will then check the number of" +
						" parameters in these parenthesis)");
	}

	@NotNull
	@Override
	public PhpElementVisitor buildVisitor(
			@NotNull final ProblemsHolder holder,
			final boolean isOnTheFly) {
		return new PhpElementVisitor() {
			@Override
			public void visitPhpNewExpression(final NewExpression expression) {
				super.visitPhpNewExpression(expression);
				final PsiElement lastChild = expression.getLastChild();
				if (lastChild == null) {
					// Typing In progress
					return;
				}
				if (!")".equals(lastChild.getText())) {
					ReinomInspection.problem(
							holder,
							expression,
							"Constructor call must use parenthesis",
							ReinomLocalQuickFix.fixByInsertingTextAfter(
									"()",
									lastChild));
				}
			}
		};
	}
}
```

(tho using the ASTNode type instead of ")".equals would be far better, but I couldn't find the right token type... :/

**Comment by orklah** [comment]
Hi,

In last version, PHPStorm added a serie of inspection to match PSR-12.

There is one of them which match that parentheses MUST be used in instanciation. It's in PHP > Code Style > PSR-12 > Missing parameter list

Maybe your issue is now covered by native PHPStorm?

**Comment by kalessil** [comment]
Can confirm, starting PS 2019.3, there's a rule disabled by default: 'PHP | Code Style | PSR-12 | Missing parameter list'. It'll cover this case 👍

**Comment by SpyroTEQ** [comment]
Ah, indeed, I didn't see this new disabled rule :) Thanks!
